src/repl.c

```diff
@@ -29,9 +49,13 @@ void	repl(const char *prompt)
 		add_history(input);
 		token_list = NULL;
 		tokenizer(input, &token_list, '\0');
-		lines = parse(token_list);
+		token_list = reverse_token_list(token_list);
+		lines = parse(token_list, env);
 		if (lines)
+		{
 			print_ast(lines->content);
+			eval(lines->content, env);
```

Comment:
```suggestion
			return_code = eval(lines->content, env);
```
or something like that.
We'd also have to pass the old return code as argument, so we can handle it in strings

Reply:
Another option would be to just add the return code to env and ignore it when printing.
That would probably easiest.